response_operations_ui/views/accounts.py

```diff
@@ -94,6 +95,53 @@ def change_account_name():
     return render_template("account/change-account-name.html", user=user, form=form, errors=form.errors)
 
 
+@account_bp.route("/change-username", methods=["GET", "POST"])
+@login_required
+def change_username():
+    form = UsernameChangeForm()
+    user_id = session["user_id"]
+    user_from_uaa = uaa_controller.get_user_by_id(user_id)
+    username = user_from_uaa["userName"]
+    username_exists = False
+    if request.method == "POST" and form.validate():
+        user_from_uaa["userName"] = form["username"].data
+        if form["username"].data != username:
+            logger.info("Sending update account details email", user_id=user_id)
+            personalisation = {
+                "first_name": user_from_uaa["name"]["givenName"],
+                "value_name": "username",
+                "changed_value": form["username"].data,
+            }
+            try:
+                NotifyController().request_to_notify(
+                    email=user_from_uaa["emails"][0]["value"],
+                    template_name="update_account_details",
+                    personalisation=personalisation,
+                )
+                uaa_errors = uaa_controller.update_user_account(user_from_uaa)
+                if uaa_errors is None:
+                    return redirect(url_for("logout_bp.logout", message="Your username has been changed"))
+                elif "response" in uaa_errors.keys() and uaa_errors["response"][0] == 400:
```

Comment:
This might need to be tweaked if you go with my suggestion for the change in format for the uaa error.